### ISSUE TITLE
Can there be a `runSync` command in this package?

### ISSUE THREAD

**Issue by therealsujitk** [tekartik/process_run.dart#103]
Dart has a [`Process.runSync()`](https://api.flutter.dev/flutter/dart-io/Process/runSync.html) command which works the same way as [`Process.run()`](https://api.flutter.dev/flutter/dart-io/Process/run.html) but synchronously. Can you add a `process_run.runSync()` command that would work the same way as [`process_run.run()`](https://pub.dev/documentation/process_run/latest/process_run.shell/run.html) but synchronously? It would also be nice if `Shell` had this method.

**Comment by alextekartik** [comment]
Good idea. Indeed it ends up being fairly easy to implement. It has some slight differences with the async versions as it won't have any `stdin` or `onProcess` parameter.

Just published in 0.14.1

## Running the script in synchronous mode

`runSync` and `runExecutableArgumentsSync` are available since 0.14.1 in the global space and in the `Shell` class.
They are synchronous version of `run` and `runExecutableArguments` respectively with some limitations.

The synchronous mode is useful for testing. It is not recommended for production use as
it is a synchronous call and will block until the child process terminates.

```dart     
var shell = Shell();
// This is a synchronous call and will block until the child process terminates.
var results = shell.runSync('echo "Hello world"');
var result = results.first;
print('output: "${result.outText.trim()}" exitCode: ${result.exitCode}');
// should display: output: "Hello world" exitCode: 0
```

**Warning**:
- You cannot feed any stdin to the child process.
- You cannot kill a synchronous child process.
- Available since 0.14.1
- Recommended for testing only, it is a synchronous call and will block until the child process terminates.

**Comment by therealsujitk** [comment]
Works perfectly! Thanks!!

**Comment by therealsujitk** [comment]
One minor thing, I think the [docs](https://pub.dev/documentation/process_run/latest/process_run.shell/runSync.html) need to be updated. The example uses `run` instead of `runSync`.

**Comment by alextekartik** [comment]
Ah yes thanks @therealsujitk . Fixed in 0.14.1+3
